kubevela/app/application.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: podtato-head
+spec:
+  components:
+    - name: podtato-head-service-discovery
+      type: k8s-objects
+      properties:
+        objects:
+          - apiVersion: v1
+            kind: ConfigMap
+            metadata:
+              name: podtato-head-discovery
+            data:
+              servicesConfig.yaml: |
+                hat:       "http://podtato-head-hat:8080"
+                left-leg:  "http://podtato-head-left-leg:8080"
+                left-arm:  "http://podtato-head-left-arm:8080"
+                right-leg: "http://podtato-head-right-leg:8080"
+                right-arm: "http://podtato-head-right-arm:8080"
+    - name: podtato-head-frontend
```

Comment:
Maybe it's better to drop the first `k8s-objects` component and make it a trait to this component like:

```
    - name: podtato-head-frontend
      type: webservice
      properties:
        image: ghcr.io/podtato-head/podtato-server:v0.3.0
        ports:
          - port: 8080
            expose: true
        env:
          - name: PODTATO_COMPONENT
            value: frontend
          - name: SERVICES_CONFIG_FILE_PATH
            value: /config/servicesConfig.yaml
        exposeType: ClusterIP
        cpu: "0.1"
        memory: "32Mi"
      traits:
        - type: scaler
          properties:
            replicas: 1
        - type: storage
          properties:
            configMap:
              - name: podtato-head-discovery
                mountPath: /config
                data: 
                  servicesConfig.yaml: |
                    hat:       "http://podtato-head-hat:8080"
                    left-leg:  "http://podtato-head-left-leg:8080"
                    left-arm:  "http://podtato-head-left-arm:8080"
                    right-leg: "http://podtato-head-right-leg:8080"
                    right-arm: "http://podtato-head-right-arm:8080"
```

Reply:
Hello @FogDong! Thank you for this comment, updated this in the scenario ...